### PR TITLE
Introducing binary compatibility tests for serialization service version 1

### PR DESCRIPTION
**Pull request by sancar**

### CODE REVIEW

**Comment by sancar** [comment]
run-lab-run

File: hazelcast/src/test/java/com/hazelcast/nio/serialization/compatibility/BinaryCompatibilityFileGenerator.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        for (Object object : objects) {
+            for (ByteOrder byteOrder : byteOrders) {
+                generateBinaryFile(outputStream, object, byteOrder);
+
+            }
+        }
+        outputStream.close();
+    }
+
+    private static String creteObjectKey(Object object, ByteOrder byteOrder) {
```
Comment (mustafaiman) [inline hazelcast/src/test/java/com/hazelcast/nio/serialization/compatibility/BinaryCompatibilityFileGenerator.java]: typo: `creteObjectKey`

**Comment by mustafaiman** [comment]
:+1:

**Comment by hasancelik** [comment]
run-lab-run

**Comment by sancar** [comment]
run-lab-run

**Comment by sancar** [comment]
run-lab-run

**Comment by sancar** [comment]
run-lab-run

**Pull request merged**